mantidproject/mantid
# Update kafka event listener to new schema

Updated the `KafkaEventListener` to use new schema for neutron event data which is designed for use by PSI, ESS and ISIS.
As part of this change the event data topic can consist of multiple partitions, allowing streaming through multiple brokers in a Kafka cluster to achieve higher throughput.

Removed mention of the Kafka listener from the release notes as it is still undergoing major changes, including to the interface. This was agreed on at the IBEX-Kafka-Mantid meeting 3 Mar 2017.

**To test:**

*To test this PR you must be at ISIS*

Launch MantidPlot and select "TEST_LIVE" as your facility and "ISIS_Kafka_Event" as your instrument. From the "Workspaces" view (crtl+shift+m) click "Load" -> "Live Data". Match the settings in the following screenshot and then hit run, after around 30 seconds the stream should end, check you have a workspace with 14258850 events in it.

![livedatatest](https://cloud.githubusercontent.com/assets/13455433/25744838/cb1c8bb6-3193-11e7-896c-a8370a5115dc.png)

Fixes #19358.

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
